drivers/digital-io/max22196/max22196.c

```diff
@@ -560,10 +567,12 @@ int max22196_remove(struct max22196_desc *desc)
 	if (!desc)
 		return -ENODEV;
 
-	for (i = 0; i < MAX22196_CHANNELS; i++) {
-		ret = max22196_set_chan_cnt(desc, i, MAX22196_CHN_CNT_RESET);
-		if (ret)
-			return ret;
+	if (desc->comm_desc) {
+		for (i = 0; i < desc->max_chn_nb; i++) {
+			ret = max22196_set_chan_cnt(desc, i, MAX22196_CHN_CNT_RESET);
+			if (ret)
+				return ret;
+		}
 	}
 
```

Comment:
please do not forget to free max22196_iio_channels, it should be done in this commit because here is where you are allocating the memory.